src/foam/nanos/ruler/RulerDAOTest.java

```diff
@@ -24,9 +24,9 @@ public void runTest(X x) {
     x = TestUtils.mockDAO(x, "ruleHistoryDAO");
 
     ruleDAO = (DAO) x.get("ruleDAO");
+    createRule(x);
     userDAO = new RulerDAO(x, (DAO) x.get("localUserDAO"), "localUserDAO");
     ruleHistoryDAO = (DAO) x.get("ruleHistoryDAO");
-    createRule(x);
```

Comment:
There will also be rules created after RulerDAO is initialized. Shouldn't we handle that?

Reply:
yes